src/foam/nanos/crunch/CapabilityAvailabilityDAO.js

```diff
@@ -96,6 +97,10 @@ foam.CLASS({
         { type: 'Capability', name: 'capability' }
       ],
       javaCode: `
+        // TODO: Rules calling capabilityDAO under the system context are
+        //   breaking availability logic. This needs proper investigation.
+        if ( ((Subject) x.get("subject")).getUser().isAdmin() ) return true;
```

Comment:
if the subject.user is admin, the permission check on line 105 would succeed, no?

Reply:
Does this apply to a system context? That's where I was finding an issue.

The problem started when I added the `IsAgent` predicate to SigningOfficer's `availabilityPredicate`, which showed me that the predicate was behaving incorrectly and getting a system context. I'll try removing this in a few minutes and see if any issues persist.

Reply:
Great - I believe the permission approach would be best suited for this (as it's currently implemented). isAdmin raises security concerns such as who can update it, when and what will it be used for and the caveat of implementing additional requirements to satisfy those concerns.